docs_sphinx/user/nmlimport.rst

```diff
@@ -0,0 +1,322 @@
+NeuroML importer (nmlimport library)
+================
+
+This is a short overview of the `~brian2tools.nmlimport` package, providing
+functionality to import Brian's morphology from a .nml file.
+
+NeuroML is a XML-based description that provides a common data format
+for defining and exchanging descriptions of neuronal cell and network models
+(`NML project website <https://neuroml.org/>`_).
+
+.. contents::
+    Overview
+    :local:
+
+Working example
+---------------
+
+As a demonstration, we will use nmlimport library to generate morphology and
+extract other related information from the ``pyramidal cell's`` nml file. You
+can download it from `here <https://github
+.com/OpenSourceBrain/ACnet2/blob/master/neuroConstruct/generatedNeuroML2/pyr_4_sym.cell.nml>`_.
+
+
+.. code:: python
+
+    from brian2tools.nmlimport.nml import NmlMorphology
+    nml_object = NmlMorphology('pyramidal.cell.nml',name_heuristic=True)
+
+This call provides us the ``nml_object`` that contains all the information
+extracted from `.nml` file. When ``name_heuristic`` param is set to True
+morphology sections will be determined based on the segment names and section
+name will be created by combining names of the inner segments of the section
+else if set to False, all linearly connected segments combines and form a
+section and naming convention `sec{unique integer}` is followed.
+
+|
+- To obtain morphology object.
+
+.. code:: python
+
+    morphology=nml_object.morphology_obj
+|
+
+- To obtain topology information.
+
+.. code:: python
+
+    print(morphology.topology())
+
+    # Output
+
+    -|  [root]
+       `-|  .apical0
+          `---|  .apical0.apical2_3_4
+          `-|  .apical0.apical1
+       `-|  .basal0
+          `-|  .basal0.basal1
+          `-|  .basal0.basal2
+|
+
+- To get the morphology area.
+
+.. code:: python
+
+    print(morphology.area)
+
+    # Output
+    [ 1228.36272755] um^2
+|
+
+- To get the morphology coordinates.
+
+.. code:: python
+
+    print(morphology.coordinates)
+
+    # Output
+    [[  0.   0.   0.]
+    [  0.  17.   0.]] um
+|
+
+- To get the morphology length.
+
+.. code:: python
+
+    print(morphology.length)
+
+    # Output
+    [ 17.] um
+|
+
+- To get the morphology distance.
+
+.. code:: python
+
+    print(morphology.distance)
+
+    # Output
+    [ 8.5] um
+|
+
+- Plot morphology using brian2tool's plot_morphology function.
+
+.. code:: python
+
+    from brian2tools.plotting.morphology import plot_morphology
+    plot_morphology(morphology)
+
+.. figure:: ../images/pyramidal_morphology.png
+   :scale: 80 %
+   :alt: Pyramidal cell's morphology plot.
+
+   Pyramidal cell's Morphology plot.
+
+
+- To get the resolved group ids of an nml `SegmentGroup`. This returns a
```

Comment:
As discussed, I think it would be good to first show here a shortened version of the NeuroML file. I'd also add a little explanation of what a `SegmentGroup` is. BTW: You should use two backticks, i.e. ` ``SegmentGroup``  ` because `SegmentGroup` is not an object defined anywhere in the documentation.